src/libraries/System.IO.FileSystem/tests/RandomAccess/FlushToDisk.cs

```diff
@@ -0,0 +1,103 @@
+            // are implemented in the base class and work by calling this "MethodUnderTest" which we override
+            // here to call the FlushToDisk() method we want to test.
+            RandomAccess.FlushToDisk(handle);
+            return 0;
+        }
+
+        [Fact]
+        public void UpdatesFileLastWriteTime()
+        {
+            // Save test file path so we can refer to the same file throughout the test.
+            string testFilePath = GetTestFilePath();
+
+            // Generate random bytes to write to file. To ensure that flushing works correctly for a variety of
+            // sizes, we want to test with buffers that are smaller than a page (e.g. just 1 byte) and buffers
+            // that are several times larger than a page (e.g. up to 10 pages).
+            int byteCount = Random.Shared.Next(1, Environment.SystemPageSize * 10);
+            byte[] randomBytes = RandomNumberGenerator.GetBytes(byteCount);
+
+            // Create a new file and open it for writing.
+            using (SafeFileHandle handle = File.OpenHandle(testFilePath, FileMode.CreateNew, FileAccess.Write))
+            {
+                // Write random bytes to file.
+                RandomAccess.Write(handle, randomBytes, fileOffset: 0);
+
+                // Get the file time BEFORE flushing to disk.
+                DateTime fileTimeBeforeFlush = File.GetLastWriteTimeUtc(testFilePath);
+
+                // Flush the file to disk. As a bare minimum test, this should work without throwing an
+                // exception on all supported platforms. Since testing that the file was actually flushed
+                // to disk is difficult without using direct/unbuffered I/O and all the challenges that come
+                // with it, we can at least test that the file time was updated and we do that later below.
+                // NOTE: the difficulty of testing this is even called out in the POSIX specification for
+                // fsync() (see https://pubs.opengroup.org/onlinepubs/009695399/functions/fsync.html).
+                RandomAccess.FlushToDisk(handle);
+
+                // Get the file time AFTER flushing to disk.
+                DateTime fileTimeAfterFlush = File.GetLastWriteTimeUtc(testFilePath);
+
+                // After explicitly flushing to disk, we expect "fileTimeAfterFlush > fileTimeBeforeFlush".
+                // However, the OS is free to flush the file to disk at any moment after the file is written
+                // to. This "implicit flush" would also update the file time and if it happens BEFORE we set the
+                // fileTimeBeforeFlush variable above, then both fileTimeBeforeFlush and fileTimeAfterFlush will
+                // end up with the same value, which is why we use the ">=" operator below instead of using ">".
+                Assert.True(fileTimeAfterFlush >= fileTimeBeforeFlush);
```

Comment:
another reason to set your own older timestamp before the flush is you eliminate this race. and then you can use Assert.NotEqual here instead. which also has the advantage that if it fails, it will be clear why, whereas when you use Assert.True it just says false expected true.

Reply:
Please see [this response](https://github.com/dotnet/runtime/pull/89100#discussion_r1273489917), I believe the decision we make there will address this too 👍